agent/src/agent/streamsets/manager.py

```diff
@@ -29,23 +27,17 @@ def delete_streamsets(streamsets_: StreamSets):
                 'Cannot move pipelines to a different streamsets as only one streamsets instance exists, cannot delete streamsets that has pipelines'
             )
         streamsets.manager.StreamsetsBalancer().unload_streamsets(streamsets_)
-    pipeline.manager.delete_monitoring_pipeline(streamsets_)
     streamsets.repository.delete(streamsets_)
 
 
 def _has_pipelines(streamsets_: StreamSets) -> bool:
-    pipelines = pipeline.repository.get_by_streamsets_id(streamsets_.id)
-    if len(pipelines) > 1:
-        return True
-    if len(pipelines) == 1 and not pipeline.manager.is_monitoring(pipelines[0]):
-        return True
-    return False
+    return True if len(pipeline.repository.get_by_streamsets_id(streamsets_.id)) >= 1 else False
 
 
 def _can_move_pipelines():
     return len(streamsets.repository.get_all()) > 1
 
-
+# TODO: pass StreamSets instance as an argument
```

Comment:
then we'll need to check pipeline_.streamsets in every place we call _client()
in order to avoid it we'll need to make major changes to streamsets manager design

Reply:
I think checking `if not pipeline_.streamsets:` is useless, agent will never have a pipeline without streamsets

Reply:
when you create a new pipeline it doesn't have streamsets until you call streamsets.manager.create(pipeline). But there are situations when you balance streamsets and you want to create a pipeline in a specific instance, so you set it to pipeline before you pass it to create()
it's not the best solution but I wasn't able to come up with a better one yet

Reply:
we can discuss it offline